### PR TITLE
[python] `string_dim_zstd_level` -> `int64_dim_zstd_level`

### PR DESCRIPTION
**Pull request by johnkerl**
## Issue and/or context:

#877

### CODE REVIEW

File: apis/python/src/tiledbsoma/dataframe.py [hunk 1]
```diff
@@ -338,7 +338,7 @@ def _build_tiledb_schema(
                 [
                     dict(
                         _type="ZstdFilter",
-                        level=tiledb_create_options.string_dim_zstd_level(),
+                        level=tiledb_create_options.int64_dim_zstd_level(),
```
Comment (bkmartinjr) [inline apis/python/src/tiledbsoma/dataframe.py:341]: the dtype of DataFrame dimensions are not hard-wired to any given primitive type - they could be string, int32, float, etc.  This really should be a more generic name.

Perhaps:  `default_datarame_dim_zstd_level`?

Comment (bkmartinjr) [reply apis/python/src/tiledbsoma/dataframe.py:341]: In keeping with that idea, maybe the config shoudl jsut have two defaults:
* DataFrame.default_dim_zstd_level
* SparseNDArray.default_dim_zstd_level

**Review by bkmartinjr — COMMENTED** [review]
nits (OK to skip if you prefer):
* missing docstrings
* missing unit test (i.e., testing the ability to change the config)

**Comment by johnkerl** [comment]
@bkmartinjr the doc-string and unit-test coverage for options issues are bigger than this PR. I've created #885, and re-opened #602, to track these.

**Review by bkmartinjr — APPROVED** [review]
LGTM

**Pull request merged**